### PR TITLE
DateTime Palette

### PR DESCRIPTION
**Pull request by Gytaco**
### Purpose

This feature integrates the Date Time ui node functionality from UI++ into the dynamo core.
https://dynamopackages.com/ search for UI++

The functionality provides a palette node to pick a Date Time from rather then typing in values.

clock

### Impact assessment

- [ ] Adds or changes a public interface (Add lukechurch to the reviewer list)
- [x] API change is added to the [API Changes document](https://github.com/DynamoDS/Dynamo/wiki/API-Changes).
- [x] Introduces new components (Add lukechurch to the reviewer list)
- [x] Changes the installer (Add sharadkjaiswal to the reviewer list)
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.

#### References 

```
Dynamo\src\Libraries\CoreNodeModels\Input\DateTimePalette.cs
Dynamo\src\Libraries\CoreNodeModelsWpf\Controls\DateTimePalette.xaml
Dynamo\src\Libraries\CoreNodeModelsWpf\NodeViewCustomizations\DateTimePalette.cs
Dynamo\test\core\dateTime\DateTimePalette.dyn
Dynamo\test\core\dateTime\DateTimePalette_BadDateTime.dyn
Dynamo\test\DynamoCoreWpfTests\DateTimePaletteTests.cs
```
Addition of Extended.WPF.Toolkit v3.0 added as installed nuget package to CoreNodeModelsWpf project.
Referencing the Xceed.Wpf.Toolkit.dll for the palette component from the package.

#### Testing strategy

To be advised.

### Review staging

- [ ] UI Review

Looks like the mono build is breaking I am going to assume this is based on using System.Windows.Media to access to bind to the object in the WPF. 
```
Dynamo\src\Libraries\CoreNodeModels\Input\DateTimePalette.cs
```

### Reviewers

TBA

### FYIs

TBA

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@kronz

File: src/Libraries/CoreNodeModels/Input/DateTimePalette.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+using SDateTime = System.DateTime;
+
+namespace CoreNodeModels.Input
+{
+    /// <summary>
+    ///     Date Time Picker node.
+    /// </summary>
+    [NodeName("DateTime Picker")]
+    //[NodeDescription("DateTimeNodeModelDescription", typeof(Resources))]
```
Comment (mjkkirschner) [inline src/Libraries/CoreNodeModels/Input/DateTimePalette.cs]: any reason these are commented out?

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/DateTimePalette.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using Dynamo.Graph.Workspaces;
+
+namespace CoreNodeModelsWpf.Nodes
+{
+    public class DateTimePaletteNodeViewCustomization : NotificationObject, INodeViewCustomization<DateTimePalette>
+    {
+        /// <summary>
+        ///     WPF Control.
+        /// </summary>
+        public DateTimePaletteUI DateTimePaletteUINode;
```
Comment (mjkkirschner) [inline src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/DateTimePalette.cs]: why is this public? can we make it private?

File: test/DynamoCoreWpfTests/DateTimePaletteTests.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+using Dynamo.Configuration;
+using Dynamo.Tests;
+using Dynamo.Graph.Workspaces;
+using DynamoCoreWpfTests.Utility;
+using NUnit.Framework;
+
+namespace DynamoCoreWpfTests
+{
+    [TestFixture]
+    class DateTimePaletteTests : DynamoModelTestBase
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/DateTimePaletteTests.cs]: could these tests be moved to DynamoCore test - is there anything here that is actually testing the wpf components or building a view?

**Comment by mjkkirschner** [comment]
Hi @Gytaco I think this looks pretty good but I have a few questions and some requests - also apologies about the long wait time.

* Can we add a test for the undo redo functionality?
* Seems it might be possible to move the tests to dynamoCore as opposed to dynamoCoreWPF?
* some info is commented out for some reason, maybe incomplete resx files?
* I don't see the nuget package you said you referenced?
* as we move towards a JSON file format new code should be added with `JSON.net` attributes or JSON converters which are capable of serializing and deserializing the node, for the date time node I think all you'll need to serialize and deserialize is the dateTime property like you are doing in XML, for json I think you need to serialize it like this:
https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14 as I think swagger uses this format.
please see other UI nodeModels in the library for examples of json attributes and potentially json constructors if necessary and let me know if you have any questions or are unable to continue this PR.

**Comment by Racel** [comment]
@Gytaco - Any updates on this? If you do not have time to make the fixes that @mjkkirschner suggested, can you please close and reopen when they can be addressed?

**Pull request closed without merging**